client/navigation/components/container/index.js

```diff
@@ -128,44 +148,18 @@ const Container = ( { menuItems } ) => {
 									category.backButtonLabel || null
 								}
 							>
-								{ !! primaryItems.length && (
-									<NavigationGroup>
-										{ primaryItems.map( ( item ) => (
-											<Item
-												key={ item.id }
-												item={ item }
-											/>
-										) ) }
-									</NavigationGroup>
+								{ renderNavigationGroup(
+									category,
+									primaryItems
 								) }
-								{ !! pluginItems.length && (
-									<NavigationGroup
-										title={
-											category.id === 'woocommerce'
-												? __(
-														'Extensions',
-														'woocommerce-admin'
-												  )
-												: null
-										}
-									>
-										{ pluginItems.map( ( item ) => (
-											<Item
-												key={ item.id }
-												item={ item }
-											/>
-										) ) }
-									</NavigationGroup>
+								{ renderNavigationGroup(
+									category,
+									pluginItems,
```

Comment:
So it looks like this will often be called with `0` items for most categories, which returns null as the first line of the function. Works well, although I wonder if we could refactor this to make the code clearer at some point? 

I don't think this has any real performance consequences so I'm not too worried in the context of this PR.

Reply:
I don't think there's any measurable performance impact, but what would you like to see to make it clearer?

We could remove the `return null` logic and instead wrap each of the groups with an item check.   This isn't very DRY, but if you feel it makes it clearer then I'm not opposed.
```js
{ primaryItems.length && renderNavigationGroup(
	category,
	primaryItems
) }
{ pluginItems.length && renderNavigationGroup(
	category,
	pluginItems,
	__( 'Extensions', 'woocommerce-admin' )
) }
{ secondaryItems.length && renderNavigationGroup(
	category,
	secondaryItems
) }
```

Reply:
Looks like this is the approach after the recent changes. Not a big deal at all, but I think this is definitely a bit clearer coming at it with no foreknowledge of what's happening. I like it :+1: